browser/src/neovim/NeovimInstance.ts

```diff
@@ -471,6 +471,11 @@ export class NeovimInstance extends EventEmitter implements INeovimInstance {
                     // set title after attaching listeners so we can get the initial title
                     await this.command("set title")
 
+                    const shell = this._configuration.getValue("terminal.shellCommand")
```

Comment:
So this is overriding the `shell` variable which currently would mislead the user thinking it's just for the terminal functionality. We have two options, to keep this behaviour but change the namespace to not include terminal or to only use the shell for the terminal functionality.

If using the second approach we would keep compatibility with the previous way, it would be a single line change to evaluate `set shell` in the `DefaultConfiguration.ts` file.

Reply:
Aw, it what this beavior I was thinking with
```
So if terminal.shellCommand is set, set shell variable to it's value
```
I wanted to align the behavior of oni's command to open terminal and :terminal command on neovim, I was surprised by the fact that the two was not equivalent

So it's like your comment, you want only to use shell as fallback :thinking: 

Having two value for this `terminal.shellCommand` for oni and `shell` for other neovim plugins seems more misleading for the user, because the shell used in oni could be different if he use a Neovim plugin that spawn a shell, it was why i wanted to aligned the two behaviors generally an user only have one shell, so why 2 different configurations ?

Reply:
Yes, sorry I meant the shell on the terminal invocation.

I agree that having multiple shell behaviours is confusing, but that's what's implied in the `terminal.shellCommand` from what I understand and that's the current behaviour.

As I mentioned, we can keep current behaviour, the only change required for that would be to fetch the `shell` variable on the default configuration. If we change it then it would be a global setting and the namespace would have to be renamed accordingly.

I think @bryphe may have an insight on this issue.

Reply:
I would actually prefer to keep the `shell` setting and `terminal.shellCommand` separate. On Windows, `set shell=powershell` doesn't work well, but I want to use `powershell` as my  terminal shell in Oni (since it has git enhancements and stuff!). So I sometimes use the `!` commands like `:!rimraf /some/dir`, and I'd like them to use `cmd.exe`, but if I open a terminal - I want powershell.

(Note that you can actually set the `shell` variable via `vim.setting.shell` today, too).

I look at `terminal.shellCommand` as just a short-hand for running `term://{some terminal command}`, and I believe there are use cases to have a separate terminal shell vs ex mode shell.

So my preference would be:
- Set `terminal.shellCommand` to null by default
- If `terminal.shellCommand` is null / undefined, and we're opening from the command palette, we just open using `term` (which will use the default shell)
- If `terminal.shellCommand` is defined, and we're opening from the command palette, we'd open with `term://${terminal.shellCommand}`

That solves my use case where I can specify 'powershell' on my windows machine for opening a terminal, but still use the default shell `cmd.exe` for `!` commands.

Reply:
(And by default, since `terminal.shellCommand` isn't defined, you get the default Vim behavior unless you explicitly override it).

Reply:
Okay, so I will implement this later (it's 2:53 for me, so it's bed time :joy: )

Reply:
Haha sounds good! That's super late, get some sleep! 😄